Source/Core/Core/PowerPC/PowerPC.cpp

```diff
@@ -173,6 +183,14 @@ void Init(int cpu_core)
     breakpoints.ClearAllTemporary();
 }
 
+void ScheduleInvalidateCacheThreadSafe(u32 address)
+{
+  if (CPU::GetState() == CPU::State::CPU_RUNNING)
+    CoreTiming::ScheduleEvent(0, s_invalidate_cache_thread_safe, address);
```

Comment:
Sorry, missed this before. You need to add `FromThread::NON_CPU` as the last argument so it will do the right thing.